tests/Microsoft.Bot.Builder.Tests/BotFrameworkAdapterTests.cs

```diff
@@ -127,11 +128,16 @@ private class MockAdapter : BotFrameworkAdapter
 
             public async override Task CreateConversationAsync(string channelId, string serviceUrl, MicrosoftAppCredentials credentials, ConversationParameters conversationParameters, BotCallbackHandler callback, CancellationToken cancellationToken)
             {
-                var activity = conversationParameters.Activity;
-                activity.ChannelData = new
-                {
-                    conversationParameters.TenantId,
-                };
+                // Create a conversation update activity to represent the result.
```

Comment:
I don't think this MockAdapter is needed.  If you construct the adapter using an HttpClient that takes a mocked HttpResponseMessage, you can let the normal adapter process the message and check results.  Maybe something like: 

```cs
        public async Task CreateConversastionOverloadProperlySetsTenantId()
        {
            // Arrange
            const string ActivityIdName = "ActivityId";
            const string ActivityIdValue = "SendActivityId";
            const string ConversationIdName = "Id";
            const string ConversationIdValue = "NewConversationId";
            const string TenantIdValue = "theTenantId";
            const string EventActivityName = "CreateConversation";

            Func<Task<HttpResponseMessage>> createResponseMessage = async () =>
            {
                var response = new HttpResponseMessage(System.Net.HttpStatusCode.OK);
                response.Content = new StringContent(new JObject { { ActivityIdName, ActivityIdValue }, { ConversationIdName, ConversationIdValue } }.ToString());
                return response;
            };

            var mockCredentialProvider = new Mock<ICredentialProvider>();
            var mockHttpMessageHandler = new Mock<HttpMessageHandler>();
            mockHttpMessageHandler.Protected()
                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
                .Returns((HttpRequestMessage request, CancellationToken cancellationToken) => createResponseMessage());

            var httpClient = new HttpClient(mockHttpMessageHandler.Object);

            var adapter = new BotFrameworkAdapter(mockCredentialProvider.Object, customHttpClient: httpClient);

            var activity = new Activity("test")
            {
                ChannelId = Channels.Msteams,
                ServiceUrl = "https://fake.service.url",
                ChannelData = new JObject
                {
                    ["tenant"] = new JObject
                    { ["id"] = TenantIdValue },
                },
                Conversation = new ConversationAccount
                { TenantId = TenantIdValue },
            };

            var parameters = new ConversationParameters()
            {
                Activity = new Activity()
                {
                    ChannelData = activity.ChannelData,
                },
            };
            var reference = activity.GetConversationReference();
            var credentials = new MicrosoftAppCredentials(string.Empty, string.Empty, httpClient);

            Activity newActivity = null;

            Task UpdateParameters(ITurnContext turnContext, CancellationToken cancellationToken)
            {
                newActivity = turnContext.Activity;
                return Task.CompletedTask;
            }

            // Act
            await adapter.CreateConversationAsync(activity.ChannelId, activity.ServiceUrl, credentials, parameters, UpdateParameters, reference, new CancellationToken());

            // Assert - all values set correctly
            Assert.AreEqual(TenantIdValue, JObject.FromObject(newActivity.ChannelData)["tenant"]["tenantId"]);
            Assert.AreEqual(ActivityIdValue, newActivity.Id);
            Assert.AreEqual(ConversationIdValue, newActivity.Conversation.Id);
            Assert.AreEqual(TenantIdValue, newActivity.Conversation.TenantId);
            Assert.AreEqual(EventActivityName, newActivity.Name);
        }

```